tools/c7n_mailer/c7n_mailer/azure/deploy.py

```diff
@@ -85,8 +85,8 @@ def provision(config):
         os.path.join(os.path.dirname(__file__), 'function.py'))
 
     packager.build(None,
-                   entry_point=os.path.join(os.path.dirname(__file__), 'handle.py'),
-                   extra_modules={'c7n_mailer', 'ruamel'})
+                   extra_modules=['c7n-mailer'],
+                   extra_non_binary_packages=['datadog', 'MarkupSafe>=0.23', 'simplejson>=3.0.0'])
```

Comment:
Same concern as previously, this seems scary but I'm not sure what a better solution is off hand.  As someone who almost weekly has to track down package version conflicts and update them I don't think I'd never know to look in here...   Almost inclined to put the array in setup.py (even if its not used there) and reference it from there :)

Reply:
i think we can utilize the same solution as above, and look at the dependencies that where installed for the package instead of hard coding here.

Reply:
Unfortunately, its not as straightforward as above solution. Above solution just takes a list of c7n & c7n_azure dependencies.
These packages can be dependencies for multiple packages (c7n_azure module dependencies) and they might have different versions requirements.. I don't want us to write code that finds suitable version.

I think easier workaround would be to get version from installed packages version.. We know these packages don't have binary wheels, so I think its safe to build a wheel from sources based on user's package version.. Any objections?

Reply:
that sounds reasonable and fine (non version dep spec here i think is what your saying, with versions picked up from installed), but curious, the transitive dependency set on azure is a bit unwieldy as a hardcode, how far down the dep tree to do you expect to look/specify in function_package.py ?